Editors/Audio/Presentation/AudioEditor/ViewModels/AudioEditorViewModel.cs

```diff
@@ -0,0 +1,291 @@
+﻿using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Windows.Input;
+using CommonControls.PackFileBrowser;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using Editors.Audio.Storage;
+using Newtonsoft.Json;
+using Shared.Core.Misc;
+using Shared.Core.PackFiles;
+using Shared.Core.PackFiles.Models;
+using Shared.Core.ToolCreation;
+using static Editors.Audio.Presentation.AudioEditor.AudioEditorViewModelHelpers;
+
+namespace Editors.Audio.Presentation.AudioEditor.ViewModels
+{
+    public class CustomStateDataGridProperties
+    {
+        public string CustomVOActor { get; set; }
+        public string CustomVOCulture { get; set; }
+        public string CustomVOBattleSelection { get; set; }
+        public string CustomVOBattleSpecialAbility { get; set; }
+        public string CustomVOFactionLeader { get; set; }
+    }
+
+    public enum DialogueEventsPreset
+    {
+        None,
+        All,
+        Essential,
+        Custom
+    }
+
+    public partial class AudioEditorViewModel : ObservableObject, IEditorViewModel
+    {
+        public readonly IAudioRepository AudioRepository;
+        public readonly PackFileService PackFileService;
+
+        public string _previousSelectedAudioProjectEvent;
+        [ObservableProperty]
+        private string _selectedAudioProjectEvent;
+        [ObservableProperty]
+        private bool _showCustomStatesOnly;
+
+        // Audio Project settings
+        [ObservableProperty]
+        private string _selectedAudioProjectEventType;
+        [ObservableProperty]
+        private string _selectedAudioProjectEventSubtype;
+        [ObservableProperty]
+        private DialogueEventsPreset _selectedAudioProjectEventsPreset;
+
+        public NotifyAttr<string> DisplayName { get; set; } = new NotifyAttr<string>("Audio Editor");
+        public AudioEditorDataViewModel AudioEditor { get; }
```

Comment:
There is a lot of public attributes here. This makes me suspect that data is acessed pleases where it should not be? 
Or that it should use the wpf community toolkit => [ObservableProperty]

Reply:
I'll look over it :)